api-audit/src/main/java/com/capitalone/dashboard/service/AuditServiceImpl.java

```diff
@@ -946,11 +904,93 @@ public boolean isGitRepoConfigured(String url, String branch) {
         }
         return false;
     }
-	
-	private List<TestResult> getTestResults(String jobUrl,long beginDt, long endDt){
-		List<TestResult> testResults = customRepositoryQuery
-				.findByUrlAndTimestampGreaterThanEqualAndTimestampLessThanEqual(jobUrl,beginDt,endDt);
-		
-		return testResults;
-	}
+
+    private List<TestResult> getTestResults(String jobUrl, long beginDt, long endDt) {
+        List<TestResult> testResults = customRepositoryQuery
+                .findByUrlAndTimestampGreaterThanEqualAndTimestampLessThanEqual(jobUrl, beginDt, endDt);
+
+        return testResults;
+    }
+
+    public PerfReviewResponse getresultsBycomponetAndTime(String businessComp, long from, long to) {
+        Cmdb cmdb = cmdbRepository.findByConfigurationItemIgnoreCase(businessComp); // get CMDB iD
```

Comment:
@sashi-kan  When I run GET validatePerfResults with businessComp name which is invalid, I get below error with nullpointer exception. Please add null check and add a message "Invalid businessComp name or does not exists" in return response. 

{"timestamp":1513021211490,"status":500,"error":"Internal Server Error","exception":"java.lang.NullPointerException","message":"No message available","path":"/apiaudit/validatePerfResults"}